src/app/students/(table)/student-table.tsx

```diff
@@ -38,10 +40,50 @@ const DataTable = <TData, TValue>({
     columns,
     data,
 }: DataTableProps<TData, TValue>) => {
-    const [rowSelection, setRowSelection] = React.useState({})
-    const [columnVisibility, setColumnVisibility] = React.useState<VisibilityState>({})
-    const [columnFilters, setColumnFilters] = React.useState<ColumnFiltersState>([])
-    const [sorting, setSorting] = React.useState<SortingState>([])
+    const [rowSelection, setRowSelection] = useState({})
+    const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+    const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+    const [sorting, setSorting] = useState<SortingState>([])
+
+    const [pagination, setPagination] = useState<PaginationState>({
+        pageIndex: 1,
+        pageSize: 4,
+    })
+
+    const [pageCount, setPageCount] = useState(0)
+    const totalStudents = useStudents()?.totalStudents ?? 0
+    const router = useRouter()
+    const pathname = usePathname()
+    const searchParams = useSearchParams()
+
+    useEffect(() => {
+        const page = parseInt(searchParams.get('page') ?? '1', 10) - 1
+        const pageSize = parseInt(searchParams.get('per_page') ?? '10', 10)
+        setPagination({
+            pageIndex: page,
+            pageSize: pageSize,
+        })
+        setPageCount(Math.ceil(totalStudents / pageSize))
+    }, [searchParams, totalStudents])
+
+    const createQueryString = (params: Record<string, string | number>) => {
+        const searchParams = new URLSearchParams()
+        Object.entries(params).forEach(([key, value]) => {
+            searchParams.set(key, value.toString())
+        })
+        return searchParams.toString()
+    }
+
+    useEffect(() => {
+        router.push(`${pathname}?${createQueryString({
+            page: pagination.pageIndex + 1,
```

Comment:
why do we add 1 here? Just make this table interpret pages the same way the API expects it. If one is 0-indexed, they both should be, if one is 1-indexed, they both should be. Unnecessary confusion to have them use different indexes and have random +1's like this

Reply:
It was tricky to do it since the react table interprets it using 0 indexed. Agree that it's confusing, will try to fix it but maybe as a seperate PR?  Also tanstack table is set on using 0 index by default, not sure how to write a good fix, see: https://github.com/TanStack/table/issues/1778

Reply:
Alright then